File: project/core/module/rest-service/src/main/java/vdi/service/rest/conversion/DatasetOwner.java
package vdi.service.rest.conversion;

import vdi.service.rest.generated.model.DatasetOwnerImpl;
import vdi.service.rest.model.UserDetails;

public class DatasetOwner extends DatasetOwnerImpl {
  public DatasetOwner(UserDetails user) {
    setUserId(user.userID().toLong());
    setFirstName(user.firstName());
    setLastName(user.lastName());
    setEmail(user.email());
    setAffiliation(user.organization());
  }
}

File: service/rest-service/src/main/kotlin/vdi/service/rest/server/outputs/DatasetOwner.kt
package vdi.service.rest.server.outputs

import vdi.service.rest.generated.model.DatasetOwner
import vdi.service.rest.generated.model.DatasetOwnerImpl
import vdi.service.model.UserDetails

fun DatasetOwner(user: UserDetails): vdi.service.rest.generated.model.DatasetOwner =
  vdi.service.rest.generated.model.DatasetOwnerImpl().apply {
    userId       = user.userID.toString().toLong()
    firstName    = user.firstName
    lastName     = user.lastName
    email        = user.email
    organization = user.organization
  }
